### PR TITLE
make generated source configurable

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: src/sbt-test/sbt-avro/settings/build.sbt [hunk 1]
```diff
@@ -7,3 +7,5 @@ libraryDependencies ++= Seq(
 
 avroStringType := "String"
 avroFieldVisibility := "public"
+(Compile / avroSource) := (Compile / sourceDirectory).value / "avro_source"
+avroGeneratedSource := sourceManaged.value / "avro"
```
Comment (RustedBones) [inline src/sbt-test/sbt-avro/settings/build.sbt]: nice settings test !

File: src/sbt-test/sbt-avro/settings/test [hunk 2]
```diff
@@ -1,5 +1,11 @@
 > compile
 
+$ exists target/scala-2.12/src_managed/main/compiled_avro/com/cavorite/test/settings/Avdl.java
+$ exists target/scala-2.12/src_managed/main/compiled_avro/com/cavorite/test/settings/Avpr.java
+$ exists target/scala-2.12/src_managed/main/compiled_avro/com/cavorite/test/settings/ProtocolAvpr.java
+$ exists target/scala-2.12/src_managed/main/compiled_avro/com/cavorite/test/settings/Avsc.java
+$ exists target/scala-2.12/src_managed/main/compiled_avro/com/cavorite/test/settings/ProtocolAvdl.java
```
Comment (RustedBones) [inline src/sbt-test/sbt-avro/settings/test]: shouldn't it be `avro` instead of `compiled_avro` ?
I think there is a problem with the setting scoping

Comment (nevillelyh) [reply src/sbt-test/sbt-avro/settings/test]: LOL good catch wonder why it didn't fail... Checking...

Comment (nevillelyh) [reply src/sbt-test/sbt-avro/settings/test]: Right it's the scoping. Fixed.

**Review by RustedBones — APPROVED** [review]
Great! This also fixes my concern in #18 . But this is the proper Fix!

**Pull request merged**